### DISCUSSION TITLE
Doris FE can't start ,because fail to find right address to start fe by using fqdn

### DISCUSSION THREAD

**Discussion by artjio** [apache/doris#38061] [A - General / Q&A] [answered]
I am using the Apache doris 2.1.4

os:ubuntu 18.0.4
java version : zulu  open jdk 1.8.0.192

When I enable the fqdn is true,the log will print "fail to find right address to start fe by using fqdn"

So,i check the fe debug log,and i found the log show doris is address my ip 196.178.5.35.

<img width="362" alt="image" src="https://github.com/user-attachments/assets/a00df093-5fe3-4358-a011-e1319091781a">

but it is weird,because my true ip is 172.23.179.53
<img width="361" alt="image" src="https://github.com/user-attachments/assets/26f30666-0a77-4432-8550-8cb0572d3ffe">

I can't find any word about 196.178.5.35. (i has been try many commend, like ifconfig,route,host,ip,nslookup)

But,if i mark  enabled_fqdn=true,doris fe will be work (log is show doris find the ip 172.23.179.53)

This is my fe.conf
<img width="349" alt="image" src="https://github.com/user-attachments/assets/d9e2e95b-2410-4b6d-af10-9c1430f0e79b">

**Comment by ixzc** [comment]
yes, if you want to start FE by using FQDN, you need to set this:
`enabled_fqdn=true`
please refer this
https://doris.apache.org/docs/dev/admin-manual/cluster-management/fqdn/

**Reply by artjio** [reply]
I'm sorry if I wasn't clear. When enable_fqdn_mode is set to true, Doris retrieves an incorrect IP and FQDN string. However, when enable_fqdn_mode is commented out, Doris can correctly retrieve the IP.

**Comment by artjio** [comment, marked as answer]
I checked the k8s configuration and found that three settings were related to the hostname. My original hostname was a string of numbers, something like 333000111. After adjusting the hostname, I noticed that the IP and FQDN retrieved by Doris changed. When I set the hostname to match the content in /etc/hosts, everything worked correctly.

k8s conf:
<img width="388" alt="image" src="https://github.com/user-attachments/assets/779adf32-5f04-46dc-8793-e5f80aa11571">

change hostname:
<img width="471" alt="image" src="https://github.com/user-attachments/assets/4cf6760a-5ee8-4a09-8d44-e7c5eb8b0b70">
